src/libraries/System.Private.CoreLib/src/System/Reflection/Assembly.cs

```diff
@@ -217,13 +217,38 @@ public override string ToString()
             return type.Module?.Assembly;
         }
 
-        // internal test hook
-        private static bool s_forceNullEntryPoint;
+        private static object? s_overriddenEntryAssembly;
+
+        /// <summary>
+        /// Sets the application's entry assembly to the provided assembly object
+        /// as argument.
+        /// </summary>
+        /// <param name="assembly">
+        /// Assembly object that represents the application's new entry assembly.
+        /// </param>
+        /// <remarks>
+        /// It is important to mention that the assembly passed to this function
+        /// has to be a runtime defined Assembly type object. Otherwise, an exception
+        /// will be thrown.
+        /// </remarks>
+        public static void SetEntryAssembly(Assembly? assembly)
+        {
+            if (assembly is null)
+            {
+                s_overriddenEntryAssembly = new object();
```

Comment:
```suggestion
                s_overriddenEntryAssembly = string.Empty;
```
This is a tiny bit more efficient placeholder.